alicloud/resource_alicloud_cs_kubernetes_autoscaler.go

```diff
@@ -0,0 +1,552 @@
+				return WrapError(fmt.Errorf("failed to update scaling group status,because of %v", err))
+			}
+
+			// get min max of scaling group
+			min, max, err := GetScalingGroupSizeRange(client, id)
+
+			if err != nil {
+				return WrapError(fmt.Errorf("failed to describe scaling group %s,because of %v", id, err))
+			}
+
+			nodeArgs := fmt.Sprintf("--nodes=%d:%d:%s", min, max, id)
+
+			args = append(args, nodeArgs)
+		}
+
+		if utilization != "" {
+			args = append(args, fmt.Sprintf("--scale-down-utilization-threshold=%s", utilization))
+			args = append(args, fmt.Sprintf("--scale-down-gpu-utilization-threshold=%s", utilization))
+		}
+
+		if coolDownDuration != "" {
+			args = append(args, fmt.Sprintf("--scale-down-delay-after-add=%s", coolDownDuration))
+			args = append(args, fmt.Sprintf("--scale-down-delay-after-failure=%s", coolDownDuration))
+		}
+
+		if deferScaleInDuration != "" {
+			args = append(args, fmt.Sprintf("--scale-down-unneeded-time=%s", deferScaleInDuration))
+		}
+
+		kubeConfPath, err := DownloadUserKubeConf(client, clusterId)
+
+		if err != nil {
+			return WrapError(fmt.Errorf("failed to download kubeconf from cluster,because of %v", err))
+		}
+
+		err = DeployAutoscaler(args, fmt.Sprintf(defaultAutoscalerImage, regionId), kubeConfPath)
+
+		if err != nil {
+			return WrapError(fmt.Errorf("failed to deploy autoscaler,because of %v", err))
+		}
+	}
+	return nil
+}
+
+// delete autoscaler deployment
+func resourceAlicloudCSKubernetesAutoscalerDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	regionId := client.RegionId
+
+	clusterId := d.Get("cluster_id").(string)
+
+	if clusterId == "" {
+		return WrapError(fmt.Errorf("please provide the cluster_id in region %s", regionId))
+	}
+
+	kubeConfPath, err := DownloadUserKubeConf(client, clusterId)
+
+	if err != nil {
+		return WrapError(fmt.Errorf("failed to download kubeconf from cluster,because of %v", err))
+	}
+
+	return DeleteAutoscaler(kubeConfPath)
+}
+
+// update scaling group config
+func UpdateScalingGroupConfiguration(client *connectivity.AliyunClient, groupId, userData string, labels string, taints string) (err error) {
+
+	describeScalingConfigurationsRequest := ess.CreateDescribeScalingConfigurationsRequest()
+	describeScalingConfigurationsResponse, err := client.WithEssClient(func(essClient *ess.Client) (interface{}, error) {
+		describeScalingConfigurationsRequest.RegionId = client.RegionId
+		describeScalingConfigurationsRequest.ScalingGroupId = groupId
+		return essClient.DescribeScalingConfigurations(describeScalingConfigurationsRequest)
+	})
+
+	configurations, ok := describeScalingConfigurationsResponse.(*ess.DescribeScalingConfigurationsResponse)
+
+	if ok != true {
+		return WrapError(fmt.Errorf("failed to parse DescribeScalingConfigurationsResponse of %s", groupId))
+	}
+
+	if err != nil {
+		return WrapErrorf(err, "failed to invoke CreateDescribeScalingConfigurations,because of %v", err)
```

Comment:
Please change the line 290 to `		return WrapErrorf(err, DefaultErrorMsg, groupId, describeScalingConfigurationsRequest.GetActionName(), AlibabaCloudSdkGoERROR)
)
`